### PR TITLE
backend: add ssh backend

### PR DESCRIPTION
**Pull request by yangdongsheng**
ssh backend to forward block request to remote disk or file
via ssh protocal.

Signed-off-by: Dongsheng Yang <dongsheng.yang@easystack.cn>

### CODE REVIEW

**Comment by yangdongsheng** [comment]
@danieldin95 Please help to review and test Thanx :)

File: ubbdadm/main.c [hunk 1]
```diff
@@ -96,6 +99,7 @@ static void usage(int status)
 		printf("\
 			ubbdadm --command map --type file --filepath PATH --devsize SIZE\n\
 			ubbdadm --command map --type rbd --pool POOL --image IMANGE \n\
+			ubbdadm --command map --type ssh --hostname HOST --filepath REMOTE_PATH --devsize SIZE --num-queues N\n\
```
Comment (danieldin95) [inline ubbdadm/main.c:102]: this not support ssh port ? default ssh port is 22, and user can assign other port like:5522.

**Review by danieldin95 — APPROVED** [review]

**Comment by danieldin95** [comment]
It's not working for ssh by pubkey, and backend has the following log:
```
2022-06-27 14:40:29.461 10504 [ERROR] main:142: failed to open backend
2022-06-27 14:41:29.583 10527 [ERROR] connect_ssh:37: Authentication failed
2022-06-27 14:41:29.584 10527 [ERROR] ssh_backend_open:64: failed to open ssh session.
2022-06-27 14:41:29.584 10527 [ERROR] main:142: failed to open backend
2022-06-27 14:42:04.189 10539 [ERROR] connect_ssh:37: Authentication failed
2022-06-27 14:42:04.190 10539 [ERROR] ssh_backend_open:64: failed to open ssh session.
2022-06-27 14:42:04.190 10539 [ERROR] main:142: failed to open backend
2022-06-27 15:27:15.338 12515 [ERROR] connect_ssh:37: Authentication failed
2022-06-27 15:27:15.338 12515 [ERROR] ssh_backend_open:64: failed to open ssh session.
2022-06-27 15:27:15.338 12515 [ERROR] main:142: failed to open backend
```

**Comment by yangdongsheng** [comment]
> It's not working for ssh by pubkey, and backend has the following log:
> 
> ```
> 2022-06-27 14:40:29.461 10504 [ERROR] main:142: failed to open backend
> 2022-06-27 14:41:29.583 10527 [ERROR] connect_ssh:37: Authentication failed
> 2022-06-27 14:41:29.584 10527 [ERROR] ssh_backend_open:64: failed to open ssh session.
> 2022-06-27 14:41:29.584 10527 [ERROR] main:142: failed to open backend
> 2022-06-27 14:42:04.189 10539 [ERROR] connect_ssh:37: Authentication failed
> 2022-06-27 14:42:04.190 10539 [ERROR] ssh_backend_open:64: failed to open ssh session.
> 2022-06-27 14:42:04.190 10539 [ERROR] main:142: failed to open backend
> 2022-06-27 15:27:15.338 12515 [ERROR] connect_ssh:37: Authentication failed
> 2022-06-27 15:27:15.338 12515 [ERROR] ssh_backend_open:64: failed to open ssh session.
> 2022-06-27 15:27:15.338 12515 [ERROR] main:142: failed to open backend
> ```

that's possible if you are using old libssh, then libssh can't open RSA public key file. So please upgrade libssh such as 0.9.3 and try again. 

Thanx

**Pull request merged**